google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -118,11 +133,34 @@ public Runnable createAutoDeploymentOperation(
         artifactToDeploy,
         defaultAppYaml(),
         defaultDockerfile(artifactType),
-        deploymentCallback
+        createCallbackWithUsageTrack(deploymentCallback, DeploymentType.AUTO, artifactType)
     );
   }
 
   @NotNull
+  private DeploymentOperationCallback createCallbackWithUsageTrack(
+      final DeploymentOperationCallback deploymentCallback,
+      DeploymentType deploymentType, DeploymentArtifactType artifactType) {
+    final String eventLabel = "deploy.flex." +
+        deploymentType.toString() + ".java" + artifactType.toString();
+
+
+    return new DeploymentOperationCallback() {
+      @Override
+      public Deployment succeeded(@NotNull DeploymentRuntime deploymentRuntime) {
+        UsageTrackerProvider.getInstance()
+            .trackEvent(GctTracking.CATEGORY, GctTracking.DEPLOY, eventLabel, null);
```

Comment:
the current breakdown for eventAction is probably not quite right.  I'm thinking we should have a GctTracking.APP_ENGINE that you pass in here. Then your event label will fit better into the hierarchy